app.rb

```diff
@@ -0,0 +1,77 @@
+require 'sinatra'
+require './memo'
+require './memo_repository'
+require './helpers/helpers'
+
+
+# Read
+
+get '/' do
+  memo_repo = MemoRepository.new
+  @files = memo_repo.find_all
+  erb :index
+end
+
+get '/show/:filename' do
+  memo_repo = MemoRepository.new
+  memo = memo_repo.find(params[:filename])
+  @filename = params[:filename]
+  @title = memo[0][0]
+  @text = memo[0][1]
```

Comment:
このままでもいいと言えばいいのですが、せっかく`Memo`と`MemoRepository`という素晴らしいクラス設計になっているので、より良いインターフェイスを目指した方がいいかなと思うのでちょっと追加でコメントを。

`memo_repo.find`メソッドを実行したら、返ってくるのは`Memo`クラスのインスタンスになるのが自然な気がします。なので、
```suggestion
  @title = memo.title
  @text = memo.text
```
という風になるのが一番直感的ではないかなと思っています。

何故なのかというのをちょっと説明してみます。
今の`find`メソッドは`CSV.read`した結果をそのまま返しちゃっているので、CSVの構造がそのまま`Memo`クラスの外に漏れ出てきてしまっているんですね。
しかし、メモがCSV形式で保存されているというのはあくまでデータの保存方法であって、この`get`アクションからするとCSVかどうかというのはどうでもいいわけです。ただメモが取得できれば良いわけですから、欲しいのは`Memo`クラスのインスタンスなのです。
例えば、19行目の処理は「変数`memo`に入っている配列の一つ目の要素の中の配列の一つ目の要素がタイトルに相当する」という知識を持っているわけですが、この知識を`Memo`クラスの中に隠蔽してしまえば、ただ`Memo.title`だけでタイトルを取り出すことができますよね。こうやって実装の詳細をクラスの中に隠蔽するのがオブジェクト指向の基本的な考え方です。

また、さらにいうと、この後CSVからDBに保存先を変えるとなったときに、今の状態だと`MemoRepository`クラスだけでなくこの19, 20行目も変更することになります。
しかし、ここが`@title = memo.title`となっていれば、変更するのは`MemoRepository`クラスだけで良くなります。保存先がCSVかDBかという詳細を知らなくてもメモを扱えるようになるというわけです。

ということで、改めて書くと、「`MemoRepository#find`メソッドが`Memo`インスタンスを返す」ようにすると非常に良い設計になりますので、是非やってみて欲しいです 🙏